src/HttpClient/Guzzle6HttpClient.php

```diff
@@ -53,6 +55,10 @@ function ($reason) {
                             throw new BadResponse($e->getMessage(), $e->getRequest(), $e->getResponse(), $e);
                         }
                     } elseif ($e instanceof RequestException) {
+                        if ($e instanceof ConnectException && CURLE_OPERATION_TIMEOUTED === ($e->getHandlerContext()['errno'] ?? null)) {
```

Comment:
`ConnectException` wins the badly named prize for the day, since it's thrown also on timeouts that happen after the connection has already been established